pages/api/changePassword.ts

```diff
@@ -0,0 +1,43 @@
+import { withApiAuth } from "../../lib/auth/withApiAuth";
+import { withApiMethods } from "../../lib/withApiMethods";
+import { withApiValidation } from "../../lib/withApiValidation";
+import { hash, compare } from "../../lib/utils/bcryptHash";
+import { prisma } from "../../lib/db";
+import { Asserts, object, string } from "yup";
+
+export const changePasswordSchema = object({
+  oldPassword: string().required("Old password is required"),
+  newPassword: string().required("New password is required"),
+  hashedOldPassword: string().required(),
+  id: string().required("User id is required"),
+});
+
+export type ChangePasswordType = Asserts<typeof changePasswordSchema>;
+
+export default withApiAuth(
+  withApiMethods({
+    POST: withApiValidation({ body: changePasswordSchema }, async (req, res) => {
+      const { oldPassword, newPassword, hashedOldPassword, id } = req.body;
+
+      if (await compare(oldPassword, hashedOldPassword)) {
+        try {
+          const hashedNewPassword = await hash(newPassword);
+
+          const changedAdmin = await prisma.admin.update({
+            where: { id },
+            data: { password: hashedNewPassword },
+          });
+
+          return res.json({
+            message: "Admin password has been changed successfully",
+            changedAdmin,
+          });
+        } catch {
+          return res.status(500).json({ message: "Internal server error" });
+        }
+      } else {
+        return res.json({ errorMessage: "Bad old password" });
```

Comment:
Let's stick to the methodology we went with earlier - the error messages are sent in the `message` field of the response.

Also, set the response status to an error one - I'd suggest `422`